### ISSUE TITLE
Search logic for installed generators - incompatible change

### ISSUE THREAD

**Issue by alex-gilin** [yeoman/environment#209]
cwd: /root/user
generators folder 'myGen': /root/somefolder/generators
/root/user/node_modules/myGen soft linked to 'myGen'

v2.8.0 
    $ cwd :: yo -> recognizes and displays installed particular generators in 'myGen'
v.2.9.1
     $ cwd :: yo -> DOES NOT recognize installed particular generators in 'myGen'

This is an incompatible change in minor version, which prevent us to use a yo@3.1.0 in our environment.

Please advise.
AG

**Comment by mshima** [comment]
All tests passed, I have to narrow down why this happened.
You can try 
```
npm install mshima/yo#master
yo --with-environment 2.8.1
```

**Comment by mshima** [comment]
Can you run and post the output?
```
DEBUG=yeoman:* yo (your parameters)
```

**Comment by alex-gilin** [comment]
As you requested :
==========================================================

user: user $ yo --version
**3.1.0**
user: user $ DEBUG=yeoman:* yo
  yeoman:environment Running lookup with options { localOnly: false, filePatterns: [ '*/index.js', 'generators/*/index.js', 'lib/generators/*/index.js' ], npmPaths: [ '/usr/lib/node_modules', '/usr/local/lib/node_modules', '/h
ome/user/.node_modules_global/lib/node_modules/yo/node_modules', '/home/user/.config/yarn/global/node_modules', '/home/user/.config/yarn/link', '/home/user/.node_modules_global/lib/node_modules', '/home/user', '/node_modules',
 '/home/node_modules', '/home/user/node_modules' ], packagePatterns: 'generator-*', packagePaths: [ '/home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates', '/home/user/node_modules/generator-basi
c-multitarget-application', '/home/user/node_modules/generator-fiori-module', '/home/user/node_modules/generator-fiori-project' ] } +0ms
  yeoman:environment Resolve namespaces for /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/index.js: @sapui5/sapui5-templates +0ms
  yeoman:environment found /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/index.js, trying to register +15ms
  yeoman:environment Registered @sapui5/sapui5-templates (/extbin/npm/globals/lib/node_modules/@sapui5/generator-sapui5-templates/generators/index.js) on package @sapui5/sapui5-templates (/home/user/.node_modules_global/lib/no
de_modules/@sapui5/generator-sapui5-templates) +0ms
  yeoman:environment Resolve namespaces for /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/1worklist/index.js: @sapui5/sapui5-templates:1worklist +1ms
  yeoman:environment found /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/1worklist/index.js, trying to register +1ms
  yeoman:environment Registered @sapui5/sapui5-templates:1worklist (/extbin/npm/globals/lib/node_modules/@sapui5/generator-sapui5-templates/generators/1worklist/index.js) on package @sapui5/sapui5-templates (/home/user/.node_m
odules_global/lib/node_modules/@sapui5/generator-sapui5-templates) +0ms
  yeoman:environment Resolve namespaces for /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/2masterdetail/index.js: @sapui5/sapui5-templates:2masterdetail +0ms
  yeoman:environment found /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/2masterdetail/index.js, trying to register +0ms
  yeoman:environment Registered @sapui5/sapui5-templates:2masterdetail (/extbin/npm/globals/lib/node_modules/@sapui5/generator-sapui5-templates/generators/2masterdetail/index.js) on package @sapui5/sapui5-templates (/home/user
/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates) +1ms
  yeoman:environment Resolve namespaces for /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/3worklistodatav4/index.js: @sapui5/sapui5-templates:3worklistodatav4 +0ms
  yeoman:environment found /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/3worklistodatav4/index.js, trying to register +1ms
  yeoman:environment Registered @sapui5/sapui5-templates:3worklistodatav4 (/extbin/npm/globals/lib/node_modules/@sapui5/generator-sapui5-templates/generators/3worklistodatav4/index.js) on package @sapui5/sapui5-templates (/hom
e/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates) +0ms
  yeoman:environment Resolve namespaces for /home/user/node_modules/generator-basic-multitarget-application/generators/app/index.js: basic-multitarget-application:app +7ms
  yeoman:environment found /home/user/node_modules/generator-basic-multitarget-application/generators/app/index.js, trying to register +7ms
  yeoman:environment Registered basic-multitarget-application:app (/extbin/generators/lib/node_modules/generator-basic-multitarget-application/generators/app/index.js) on package basic-multitarget-application (/home/user/node_
modules/generator-basic-multitarget-application) +0ms
  yeoman:environment Resolve namespaces for /home/user/node_modules/generator-fiori-module/generators/app/index.js: fiori-module:app +7ms
  yeoman:environment found /home/user/node_modules/generator-fiori-module/generators/app/index.js, trying to register +7ms
  yeoman:environment Registered fiori-module:app (/extbin/generators/lib/node_modules/generator-fiori-module/generators/app/index.js) on package fiori-module (/home/user/node_modules/generator-fiori-module) +0ms
  yeoman:environment Resolve namespaces for /home/user/node_modules/generator-fiori-project/generators/app/index.js: fiori-project:app +7ms
  yeoman:environment found /home/user/node_modules/generator-fiori-project/generators/app/index.js, trying to register +7ms
  yeoman:environment Registered fiori-project:app (/extbin/generators/lib/node_modules/generator-fiori-project/generators/app/index.js) on package fiori-project (/home/user/node_modules/generator-fiori-project) +0ms
? 'Allo! What would you like to do? (Use arrow keys)
  Run a generator
❯ Basic Multitarget Application 
  Fiori Module 
  Fiori Project 
  ──────────────
  Update your generators 
  Install a generator 
(Move up and down to reveal more choices)

==================================================================================================================================================================================================================================  
  
  user: vault $ yo --version
**3.1.1**
user: vault $ DEBUG=yeoman:* yo
  yeoman:environment:repository Running npm with args [ 'root', '-g', '--prefix', '/home/user/projects/vault/.yo-repository', '--loglevel', 'error' ] +0ms
  yeoman:environment Running lookup with options: { localOnly: false, filePatterns: [ '*/index.ts', '*/index.js', 'generators/*/index.ts', 'generators/*/index.js', 'lib/generators/*/inde
x.ts', 'lib/generators/*/index.js' ], filterPaths: false, packagePatterns: 'generator-*', globbyDeep: 1, reverse: true } +0ms
  yeoman:environment Lookup calculated options: { localOnly: false, filePatterns: [ '*/index.ts', '*/index.js', 'generators/*/index.ts', 'generators/*/index.js', 'lib/generators/*/index.
ts', 'lib/generators/*/index.js' ], filterPaths: false, packagePatterns: 'generator-*', globbyDeep: 1, reverse: true, npmPaths: [ '/usr/lib/node_modules', '/usr/local/lib/node_modules', 
'/home/user/.node_modules_global/lib/node_modules/yo/node_modules', '/home/user/.config/yarn/global/node_modules', '/home/user/.config/yarn/link', '/home/user/.node_modules_global/lib/no
de_modules', '/home/user', '/node_modules', '/home/node_modules', '/home/user/node_modules', '/home/user/projects/node_modules', '/home/user/projects/vault/node_modules' ], packagePaths:
 [ '/home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates' ] } +441ms
  yeoman:environment Resolve namespaces for /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/index.js: @sapui5/sapui5-templates +0ms
  yeoman:environment found /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/index.js, trying to register +21ms
  yeoman:environment Registered @sapui5/sapui5-templates (/extbin/npm/globals/lib/node_modules/@sapui5/generator-sapui5-templates/generators/index.js) on package @sapui5/sapui5-templates
 (/home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates) +1ms
  yeoman:environment Resolve namespaces for /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/1worklist/index.js: @sapui5/sapui5-templates:1w
orklist +0ms
  yeoman:environment found /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/1worklist/index.js, trying to register +2ms
  yeoman:environment Registered @sapui5/sapui5-templates:1worklist (/extbin/npm/globals/lib/node_modules/@sapui5/generator-sapui5-templates/generators/1worklist/index.js) on package @sap
ui5/sapui5-templates (/home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates) +1ms
  yeoman:environment Resolve namespaces for /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/2masterdetail/index.js: @sapui5/sapui5-template
s:2masterdetail +0ms
  yeoman:environment found /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/2masterdetail/index.js, trying to register +0ms
  yeoman:environment Registered @sapui5/sapui5-templates:2masterdetail (/extbin/npm/globals/lib/node_modules/@sapui5/generator-sapui5-templates/generators/2masterdetail/index.js) on pack
age @sapui5/sapui5-templates (/home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates) +0ms
  yeoman:environment Resolve namespaces for /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/3worklistodatav4/index.js: @sapui5/sapui5-templ
ates:3worklistodatav4 +1ms
  yeoman:environment found /home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates/generators/3worklistodatav4/index.js, trying to register +1ms
  yeoman:environment Registered @sapui5/sapui5-templates:3worklistodatav4 (/extbin/npm/globals/lib/node_modules/@sapui5/generator-sapui5-templates/generators/3worklistodatav4/index.js) o
n package @sapui5/sapui5-templates (/home/user/.node_modules_global/lib/node_modules/@sapui5/generator-sapui5-templates) +0ms
? 'Allo! What would you like to do? (Use arrow keys)
  Run a generator
  ──────────────
❯ Install a generator 
  Find some help 
  Get me out of here! 
  ──────────────


You can see the output for both correct (yo@3.1.0 with yeoman-environment 2.8.0) and incorrect (yo@3.1.1 with yeoman-environment 2.9.2).
Please advice.

**Comment by mshima** [comment]
@alex-gilin the entire node_modules is linked?

**Comment by alex-gilin** [comment]
I think so.

**Comment by mshima** [comment]
Ok then.
There are test for node_modules with linked generator.
I don't think there are test for linked node_modules.

**Comment by mshima** [comment]
Should be fixed in 2.9.4
